src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -445,7 +448,9 @@ public void run() {
                     if (!channel.isConnected() || !sshConnection.isConnected()) {
                         throw new IllegalStateException("SSH connection is already lost.");
                     }
-                    sleep(SSH_RX_SLEEP_MILLIS);
+                    if (linecount > 0) {
```

Comment:
Probably I'm a bit too paranoid about this performance bottleneck. I encountered it while running the unit tests with a buffer size of 12, when suddently the tests started failing but output was correct. It just took too long to read everything while the unit tests started validating the calls.
This will only be a bottleneck, when we have events coming in so that the buffer is filled faster than is is emptied. That would mean with a 16kb buffer that events come in with more than 160kb/s. I guess this is rarely happening.
Personal impression is that event size is usually less than 1kb on our gerrit. But we sometimes have 7 or 8 events coming in basically at the same time, so they are all read by reader.read(cb) together.
In [JENKINS-44568](https://issues.jenkins-ci.org/browse/JENKINS-44568) it is mentioned that on gerrithub events were up to 78kb.
Maybe it is sufficient to make the buffer size configurable and mention this potential bottleneck.
What I don't know is how big the buffers of the InputStream and the Socket are. If the Socket buffer is full, gerrit side will hang up.

Maybe use 
if (readCount == 0 || readCount > 0 && linecount > 0) 
This way we would not sleep only when we have read data but not encountered a newline.